tests/unit/util/test.inc

```diff
@@ -464,6 +464,62 @@ static void test_mixing_GET_ARG_and_FOR_EACH(void)
 	zassert_equal(a[4], 5, NULL);
 }
 
+static void test_PART_OF_ARRAY(void)
+{
+	size_t i;
+	size_t array[3];
+	uint8_t *const alias = (uint8_t *)array;
+
+	zassert_false(PART_OF_ARRAY(array, &array[-1]), NULL);
+	zassert_false(PART_OF_ARRAY(array, &array[ARRAY_SIZE(array)]), NULL);
```

Comment:
Could we add a test where we do `PART_OF_ARRAY` on an field in a struct, and expect it to fail? 

Ditto with adding the same test for `test_ARRAY_INDEX_FLOOR`, but where it will succeed?

Reply:
> Could we add a test where we do `PART_OF_ARRAY` on an field in a struct, and expect it to fail?

It's already present (below), but it is expected to pass for `PART_OF_ARRAY` (in contrast to `IS_ARRAY_ELEMENT`).

Note, I'm not explicitly using a field in a struct, but simply an address that is not aligned to an array-element boundary (`alias`), which is equivalent.

> Ditto with adding the same test for `test_ARRAY_INDEX_FLOOR`, but where it will succeed?

Already done (same as above).

Reply:
Ah, sorry, missed that (`alias` and `array` were apparently too similar for my simple eyes to notice :D )